### ISSUE TITLE
Lint/AssignmentInCondition regression causing false positive

### ISSUE THREAD

**Issue by savef** [bbatsov/rubocop#3520]
The following code will raise an offence for the Lint/AssignmentInCondition cop:
`foo if bar { |x| x.abc = nil }.value`

This didn't used to happen in v0.42.0, but does with v0.43.0. I tracked the commit that causes the breaking change to b0d8e0cf.

-----
## RuboCop version
```
$ rubocop -V
0.43.0 (using Parser 2.3.1.3, running on ruby 2.3.1 x86_64-linux)
```

**Comment by savef** [comment]
@Drenmi

**Comment by Drenmi** [comment]
Thanks for the report @savef! This would suggest we're missing some test cases. 😀 

I will have a look at this as soon as I have time. If someone else wants to pick it up before that may happen, please feel free.

**Comment by savef** [comment]
Sorted.
